lua/telescope/actions/init.lua

```diff
@@ -1055,6 +1057,260 @@ actions.which_key = function(prompt_bufnr, opts)
   end
 end
 
+local os_sep = Path.path.sep
+
+local get_marked_files = function(prompt_bufnr, smart)
+  smart = vim.F.if_nil(smart, true)
+  local selected = {}
+  action_utils.map_selections(prompt_bufnr, function(entry)
+    return table.insert(selected, Path:new(entry[1]))
+  end, smart)
+  return selected
+end
+
+local is_dir = function(value)
+  return string.sub(value, -1, -1) == os_sep
+end
+
+--- Creates a new file in the current directory of the |builtin.file_browser|.
+--- Notes:
+--- - You can create folders by ending the name in the path separator of your OS, e.g. "/" on Unix systems
+--- - You can implicitly create new folders by passing $/CWD/new_folder/filename.lua
+---@param prompt_bufnr number: The prompt bufnr
+actions.create_file = function(prompt_bufnr)
+  local current_picker = action_state.get_current_picker(prompt_bufnr)
+  local finder = current_picker.finder
+  local file = action_state.get_current_line()
+  if file == "" then
+    print(
+      "To create a new file or directory(add "
+        .. os_sep
+        .. " at the end of file) "
+        .. "write the desired new into the prompt and press <C-e>. "
+        .. "It works for not existing nested input as well."
+        .. "Example: this"
+        .. os_sep
+        .. "is"
+        .. os_sep
+        .. "a"
+        .. os_sep
+        .. "new_file.lua"
+    )
+    return
+  end
+  local fpath = Path:new(finder.path .. os_sep .. file)
+  if fpath:exists() then
+    print "File or folder already exists."
```

Comment:
I believe in that case it might be better to use `vim.api.nvim_echo` with Error highlight.